django/db/backends/postgresql/operations.py

```diff
@@ -43,6 +43,7 @@ class DatabaseOperations(BaseDatabaseOperations):
         "AutoField": "integer",
         "BigAutoField": "bigint",
         "SmallAutoField": "smallint",
+        "CharField": cast_char_field_without_max_length,
```

Comment:
This is not an acceptable solution, we cannot regress `Cast()` on PostgreSQL because it's used somewhere else (in this case in the `bulk_update()`).

Reply:
I don't think we are regressing the `Cast()` here the query generated is 
```UPDATE "queries_article" SET "name" = (CASE WHEN ("queries_article"."id" = 1) THEN bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb WHEN ("queries_article"."id" = 2) THEN bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ELSE NULL END)::varchar WHERE "queries_article"."id" IN (1, 2)``` 
it is just generating the super type i.e. `varchar` instead of `varchar(20)` in case of a `CharField`

If you can explain a little further that would be great @felixxm

Reply:
If we don't regress then why you skipped `test_cast_to_char_field_with_max_length` on PostgreSQL?

Reply:
As far as I'm aware PostgreSQL supports casting `varchar` with a precision, so why you want to add `supports_cast_char_field_with_precision`  and set it to `False` on PostgreSQL?

Reply:
Got it, so the behavior of {{{Cast}}} has to remain unchanged.

Reply:
> As far as I'm aware PostgreSQL supports casting `varchar` with a precision, so why you want to add `supports_cast_char_field_with_precision` and set it to `False` on PostgreSQL?

I think I completely ignored the casting angle here, just focused on the bulk update only. But I think the direction is right, we can do these changes in a subclass of Cast and call that in bulk update